### PR TITLE
chore: remove 'deploy-github' script and improve setup docs

### PR DESCRIPTION
**Pull request by skaldarnar**
Starting the clean up by looking at the scripts defined in `package.json` and removing redundant scripts, legacy code, and improving the documentation.

The deploy script is no longer needed, as deployment is handled via `.github/deploy.yml`.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -6666,18 +6623,6 @@ getopts@^2.1.1:
...
-  version "2.2.0"
-  resolved "https://registry.yarnpkg.com/gh-pages/-/gh-pages-2.2.0.tgz#74ebeaca8d2b9a11279dcbd4a39ddfff3e6caa24"
-  integrity sha512-c+yPkNOPMFGNisYg9r4qvsMIjVYikJv7ImFOhPIVPt0+AcRUamZ7zkGRLHz7FKB0xrlZ+ddSOJsZv9XAFVXLmA==
-  dependencies:
-    async "^2.6.1"
-    commander "^2.18.0"
-    email-addresses "^3.0.1"
-    filenamify-url "^1.0.0"
-    fs-extra "^8.1.0"
-    globby "^6.1.0"
```
Comment (skaldarnar) [inline yarn.lock:6679]: this and dependencies only used by `gh-pages` was removed from the lockfile :upside_down_face:

File: package.json [hunk 2]
```diff
@@ -78,14 +77,10 @@
...
     "develop": "gatsby develop",
-    "dev": "npm run develop",
     "serve": "gatsby serve",
     "build": "gatsby build",
     "build:pp": "gatsby build --prefix-paths",
-    "build:gh": "npm run clean && npm run build:pp && gh-pages -d public",
-    "clean": "rm -rf public && rm -rf .cache",
-    "deploy": "gatsby build && node ./scripts/deploy-github",
-    "render": "node ./scripts/deploy-render && gatsby build",
+    "clean": "gatsby clean",
```
Comment (skaldarnar) [inline package.json:83]: This seems to be a saner way to clean the outputs of Gatsby than `rm -f`-ing some directories.

File: package.json [hunk 3]
```diff
@@ -33,7 +33,6 @@
     "gatsby-transformer-remark": "^4.4.1",
     "gatsby-transformer-sharp": "^3.7.1",
     "gbimage-bridge": "^0.1.4",
-    "gh-pages": "^2.0.1",
```
Comment (skaldarnar) [inline package.json:36]: Was only used by the script I removed.

File: README.md [hunk 4]
```diff
@@ -29,42 +29,52 @@ The module website development environment can be installed on macOS, Windows, a
 
 ---
 
-<h2 id="built-with">Built With</h2>
+## Prerequisites
 
-The following generator is built using the following software, you'll need them installed in your workspace to run properly. 
-* [Node](https://nodejs.org/en/)
-* [Gatsby](https://www.gatsbyjs.org/)
```
Comment (skaldarnar) [inline README.md:36]: We are building with Gatsby, but it's actually not required to have it installed, as it is managed as a dependency via Yarn.

File: README.md [hunk 5]
```diff
@@ -29,42 +29,52 @@ The module website development environment can be installed on macOS, Windows, a
...
+* [Node.js](https://nodejs.org/en/)
 * [Yarn](https://yarnpkg.com/en/)
 
 ---
 
-<h2 id="setup"> Setting Up Development Workspace </h2>
+## Development
 
-Start by cloning your fork of the ModuleSite repository and connecting the ModuleSite upstream repository:
+Make sure that you fulfil the [prerequisites](#prerequisites) for building the website locally
```
Comment (jdrueckert) [inline README.md]:
```suggestion
Make sure that you fulfill the [prerequisites](#prerequisites) for building the website locally
```

**Review by jdrueckert — DISMISSED** [review]

**Review by jdrueckert — APPROVED** [review]

**Pull request merged**